storage/tests/unit/test_client.py

```diff
@@ -441,6 +476,39 @@ def test_create_bucket_success(self):
         json_sent = http.request.call_args_list[0][1]["data"]
         self.assertEqual(sent, json.loads(json_sent))
 
+    def test_create_bucket_with_object_success(self):
+        from google.cloud.storage.bucket import Bucket
+
+        PROJECT = "PROJECT"
+        CREDENTIALS = _make_credentials()
+        client = self._make_one(project=PROJECT, credentials=CREDENTIALS)
+
+        BUCKET_NAME = "bucket-name"
+        bucket_obj = Bucket(client, BUCKET_NAME)
+        URI = "/".join(
+            [
+                client._connection.API_BASE_URL,
+                "storage",
+                client._connection.API_VERSION,
+                "b?project=%s" % (PROJECT,),
+            ]
+        )
+        sent = {"name": BUCKET_NAME, "billing": {"requesterPays": True}}
```

Comment:
In addition to `requesterPays`, let's set some properties on the bucket object in this test. The properties I most want to see are

1. ~[location](https://googleapis.github.io/google-cloud-python/latest/storage/buckets.html#google.cloud.storage.bucket.Bucket.location) - Which region is this bucket belonging to?~ (actually, that's not settable yet. Let's add that to the list in the redesign doc if it's not there yet)
2. [`storage_class`](https://googleapis.github.io/google-cloud-python/latest/storage/buckets.html#google.cloud.storage.bucket.Bucket.storage_class) - What kind of bucket is this?
3. [`requester_pays`](https://googleapis.github.io/google-cloud-python/latest/storage/buckets.html#google.cloud.storage.bucket.Bucket.requester_pays) - Leave the argument to `create_bucket` unset and set it on the Bucket, instead. (We should deprecate `requester_pays` on `create_bucket`. Add that to the redesign list, too.)

Reply:
Added storage_class property and adjusted requester_pays as directed. Included requester_pays and location comment in the redesign doc.